src/plugins/irc-events/join.js

```diff
@@ -17,11 +18,15 @@ module.exports = function(irc, network) {
 				chan: chan
 			});
 		}
-		chan.users.push(new User({nick: data.nick, modes: ""}));
-		chan.sortUsers(irc);
-		client.emit("users", {
-			chan: chan.id
-		});
+
+		if (!_.find(chan.users, {nick: data.nick})) {
```

Comment:
This seems like a poor man's fix to me, you know one of these fixes trying to fix the symptom without tackling the root cause itself.
I'm not sure why this check is necessary, could you add a comment here so we don't scratch our head 6 months from now?

Also, why is this necessary here and not in the `client.emit("join" ...` or the `chan.pushMessage(client, msg)`?